sqlx-core/src/any/connection/mod.rs

```diff
@@ -39,6 +39,20 @@ impl AnyConnection {
         })
     }
 
+    /// UNSTABLE: for use with `sqlx-cli`
+    ///
+    /// Connect to the database, and instruct the nested driver to
+    /// read options from the sqlx.toml file as appropriate.
+    #[doc(hidden)]
+    pub fn connect_with_config(url: &str) -> BoxFuture<'static, Result<Self, Error>>
```

Comment:
Would it be reasonable to make this available to apps using sqlx other than just the cli? What makes it a sqlx-cli only option? Would it make sense to instead make loading the config file default functionality (or perhaps feature gated instead of adding a new method)?

Reply:
Perhaps, but I was working on the principle of least change. The CLI is the only place I needed it. It would not be tagged as pub if that weren't a technical necessity, so marking it unstable and hiding it in the docs is the next best option.